src/base/utils/version.h

```diff
@@ -133,19 +133,15 @@ namespace Utils
             return (*this != ThisType {});
         }
 
-        constexpr bool operator==(const ThisType &other) const
+        // TODO: remove manually defined operators and use compiler generated `operator<=>()` in C++20
+        friend bool operator==(const ThisType &left, const ThisType &right)
         {
-            return (m_components == other.m_components);
+            return (left.m_components == right.m_components);
         }
 
-        constexpr bool operator<(const ThisType &other) const
+        friend bool operator<(const ThisType &left, const ThisType &right)
```

Comment:
Don't you also need `operator>`?

Reply:
And why do you remove `constexpr` specifier?

Reply:
>Don't you also need operator>?

See line 198.

>And why do you remove constexpr specifier?

It is not really supported in C++14, see https://en.cppreference.com/w/cpp/container/array/operator_cmp

Reply:
@Chocobo1 @glassez 

> It is not really supported in C++14, see https://en.cppreference.com/w/cpp/container/array/operator_cmp

Not to derail the conversation too much in this direction, but since this is being brought up: When do you think it is convenient to transition to C++17? In your opinion, is it something that should only done as part of the proper transition to Qt6? It could be done sooner, if convenient; all platforms that we support have easy access to toolchains with feature-complete (minus a couple of minor footnotes that are not really relevant for us, if we're being really pedantic) C++17 support (for example, GCC 9.x is available even in the official repos of Ubuntu 18.04). As far as the buildsystem goes, for CMake its only a matter of setting the compile features to C++17 in the CML and it will automagically insert the appropriate `--std=` flags where needed.

Reply:
>When do you think it is convenient to transition to C++17? 

As soon as I got all smaller things done, then I would propose it myself.

>In your opinion, is it something that should only done as part of the proper transition to Qt6? 

Yes.